sources/common/infoschema.go

```diff
@@ -162,8 +162,19 @@ func processTable(conv *internal.Conv, db *sql.DB, table SchemaAndName, infoSche
 	}
 	name := infoSchema.GetTableName(table.Schema, table.Name)
 	var schemaPKeys []schema.Key
-	for _, k := range primaryKeys {
-		schemaPKeys = append(schemaPKeys, schema.Key{Column: k})
+	if len(primaryKeys) == 0 {
+		for col, constraintList := range constraints {
+			for _, constraint := range constraintList {
+				if constraint == "UNIQUE" {
+					schemaPKeys = append(schemaPKeys, schema.Key{Column: col})
+					break
+				}
```

Comment:
This piece currently takes all cols with UNIQUE as part of primary key. Should we just take one rather than all?